governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -0,0 +1,210 @@
+            }
+        }
+    }
+
+    private List<Class<? extends Module>> getModulesForAnnotatedClass(Class<?> testClass) {
+        final Annotation annotation = testClass.getAnnotation(ModulesForTesting.class);
+        if (annotation != null) {
+            return Arrays.asList(((ModulesForTesting) annotation).value());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    private void getMocksForTestClass(Class<?> testClass) {
+        getMocksForAnnotatedFields(testClass);
+        for (Class<?> parentClass : ClassUtils.getAllSuperclasses(testClass)) {
+            getMocksForAnnotatedFields(parentClass);
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    private void getMocksForAnnotatedFields(Class<?> testClass) {
+
+        for (Field field : testClass.getDeclaredFields()) {
+            if (field.isAnnotationPresent(ReplaceWithMock.class)) {
+                overrideModules.add(new MockitoOverrideModule(field.getAnnotation(ReplaceWithMock.class), field.getType()));
+            }
+            if (field.isAnnotationPresent(WrapWithSpy.class)) {
+                spyTargets.add(Key.get(field.getType()));
+            }
+        }
+    }
+
+    private void getSpiesForTargetKeys(List<Element> elements) {
+        for (Element element : elements) {
+            element.acceptVisitor(new DefaultElementVisitor<Void>() {
+                @Override
+                public <T> Void visit(Binding<T> binding) {
+                    final Binding<T> finalBinding = binding;
+                    if (spyTargets.contains(binding.getKey())) {
+                        AbstractModule spyModule = new AbstractModule() {
+                            protected void configure() {
+
+                                final Key newUniqueKey = Key.get(finalBinding.getKey().getTypeLiteral().getRawType(),
+                                        Names.named("Spied " + finalBinding.getKey().getTypeLiteral().getRawType()));
+                                finalBinding.acceptTargetVisitor(new CopyBindingTargetVisitor<>(binder().bind(newUniqueKey)));
+                                bind(finalBinding.getKey()).toProvider(new SingletonProvider<T>() {
+                                    @Inject
+                                    Injector injector;
+
+                                    protected T create() {
+                                        T t = (T) injector.getInstance(newUniqueKey);
+                                        return Mockito.spy(t);
+                                    };
+                                });
+                            }
+                        };
+                        overrideModules.add(spyModule);
+                    }
+                    return null;
+                }
+            });
+        }
+    }
+    
+    public void prepareConfigForTestClass(Class<?> testClass, Method testMethod) {
+        for(Class<?> parentClass : ClassUtils.getAllSuperclasses(testClass)) {
```

Comment:
probably should iterate through the superclasses in reverse order, so that properties from the most senior parent are overlaid by those of the more junior parent